### PR TITLE
Kovacz

### PR DESCRIPTION
**Pull request by akukh**

### CODE REVIEW

File: KovaczListOfStrings/list.h [hunk 1]
```diff
@@ -0,0 +1,214 @@
+#pragma once
+#include <stdlib.h>
+#include <string.h>
+
```
Comment (dnchk) [inline KovaczListOfStrings/list.h]: In general, every .cpp file should have an associated .h file.

File: KovaczListOfStrings/list.h [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+#define DATA(list) (list[0])
```
Comment (dnchk) [inline KovaczListOfStrings/list.h]: In general macros should not be used. However, if they are absolutely needed, then they should be named with all capitals and underscores.

File: KovaczListOfStrings/list.h [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+	while (NEXT(list_ptr) != last_element)
+	{
+		curr_element = list_ptr;
+		while (NEXT(curr_element) != last_element)
+		{
+			next_element = NEXT(curr_element);
+			if (strlen(DATA(curr_element)) > strlen(DATA(next_element)))
+			{
+				char_ptr tmp		=	DATA(next_element);
```
Comment (dnchk) [inline KovaczListOfStrings/list.h]: Avoid generic names, like tmp and retval.

File: KovaczListOfStrings/main.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include "list.h"
```
Comment (dnchk) [inline KovaczListOfStrings/main.cpp]: Order your includes in header file as follows:
1. System files.
2. Other 3-rd party libraries' .h files.
3. Your project's .h files.

File: KovaczListOfStrings/main.cpp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	list_element str_list = NULL;
+
+	char str1[256];
+	char str2[256];
+	char str3[256];
+	gets(str1);
+	gets(str2);
+	gets(str3);
+
+	str_list_add(&str_list, str1);
```
Comment (dnchk) [inline KovaczListOfStrings/main.cpp]: Correct function name is StringListAdd()

**Review by dnchk — CHANGES_REQUESTED** [review]
Some comments about your code.su

File: KovaczListOfStrings/list.h (hunk 2 above)
Comment (akukh) [reply KovaczListOfStrings/list.h]: Main key should not be, but actually can be, in my case its absolutely needed, and justified

File: KovaczListOfStrings/list.h (hunk 3 above)
Comment (akukh) [reply KovaczListOfStrings/list.h]: In my case its justified bec its really temporal variable

File: KovaczListOfStrings/main.cpp (hunk 4 above)
Comment (akukh) [reply KovaczListOfStrings/main.cpp]: ???

File: KovaczListOfStrings/list.h (hunk 1 above)
Comment (akukh) [reply KovaczListOfStrings/list.h]: every .cpp file should have an associated .h file but not vise versa
i dont have list.cpp

File: KovaczListOfStrings/main.cpp (hunk 4 above)
Comment (dnchk) [reply KovaczListOfStrings/main.cpp]: You included your .h file "list.h" before <stdio.h>

**Review by dnchk — CHANGES_REQUESTED** [review]

File: KovaczListOfStrings/main.cpp (hunk 4 above)
Comment (akukh) [reply KovaczListOfStrings/main.cpp]: Ofc, bec <stdio.h>is located in Preprocessor block and and is used only for debugging.
Карочі, можна канєшно розділити і винести <stdio.h> вверх але тоді буде щось типу 
#ifdef DEBUG
#include <stdio.h>
#endif // DEBUG
мені здається це лише знизить читабельність, ну і сама #define DEBUG оприділена нижче.
Я думаю для дебаг версії можна допустити таку мікро помилку?

**Pull request closed without merging**